homeassistant/components/climate/mill.py

```diff
@@ -0,0 +1,158 @@
+"""
+Support for mill wifi-enabled home heaters.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.mill/
+"""
+
+import logging
+
+import voluptuous as vol
+from homeassistant.components.climate import (
+    ClimateDevice, PLATFORM_SCHEMA, SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_FAN_MODE, SUPPORT_ON_OFF)
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_EMAIL, CONF_PASSWORD,
+    STATE_ON, STATE_OFF, TEMP_CELSIUS)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+REQUIREMENTS = ['millheater==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+MAX_TEMP = 35
+MIN_TEMP = 5
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE |
+                 SUPPORT_FAN_MODE | SUPPORT_ON_OFF)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): cv.string,
```

Comment:
We most often use `CONF_USERNAME` even for email username. But you decide what fits best.